### PR TITLE
release spark operator chart 1.1.17

### PR DESCRIPTION
**Pull request by cwyl02**
https://jira.d2iq.com/browse/D2IQ-84004

### CODE REVIEW

File: d2iq/kuttl-test-templates/manifests/install-volcano-all-in-one.yaml [hunk 1]
```diff
@@ -7609,6 +7609,9 @@ rules:
   - apiGroups: ["nodeinfo.volcano.sh"]
     resources: ["numatopologies"]
     verbs: ["get", "list", "watch", "delete"]
+  - apiGroups: ["storage.k8s.io"]
```
Comment (cwyl02) [inline d2iq/kuttl-test-templates/manifests/install-volcano-all-in-one.yaml:7612]: manually highlight changes that I added

File: pkg/webhook/webhook.go [hunk 2]
```diff
@@ -28,10 +28,8 @@ import (
 
 	"github.com/golang/glog"
 
-	admissionv1beta1 "k8s.io/api/admission/v1beta1"
-	"k8s.io/api/admissionregistration/v1beta1"
-	arv1beta1 "k8s.io/api/admissionregistration/v1beta1"
-	apiv1 "k8s.io/api/core/v1"
+	admissionv1 "k8s.io/api/admission/v1"
```
Comment (cwyl02) [inline pkg/webhook/webhook.go:31]: changes that makes things work in k8s 1.22

File: charts/spark-operator-chart/values.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ image:
   # -- Image pull policy
   pullPolicy: IfNotPresent
   # -- Overrides the image tag whose default is the chart appVersion.
-  tag: "3.1.1-1.1.6"
+  tag: "3.1.1-1.1.17"
```
Comment (cwyl02) [inline charts/spark-operator-chart/values.yaml:15]: manually highlight changes that I added. image has been retagged

File: pkg/controller/sparkapplication/controller.go [hunk 4]
```diff
@@ -442,7 +442,9 @@ func (c *Controller) getAndUpdateAppState(app *v1beta2.SparkApplication) error {
 }
 
 func (c *Controller) handleSparkApplicationDeletion(app *v1beta2.SparkApplication) {
-	c.metrics.exportMetricsOnDelete(app)
+	if c.metrics != nil {
```
Comment (cwyl02) [inline pkg/controller/sparkapplication/controller.go:445]: manually highlight changes that I added. this fixes the nil pointer on SparkApp deletion

**Review by kaiwalyajoshi — APPROVED** [review]

**Pull request merged**